pythia6.sh

```diff
@@ -33,6 +33,7 @@ module load BASE/1.0
 # Our environment
 setenv PYTHIA6_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
 prepend-path LD_LIBRARY_PATH \$::env(PYTHIA6_ROOT)/lib
+prepend-path AGILE_GEN_PATH \$::env(PYTHIA6_ROOT)
 $([[ ${ARCHITECTURE:0:3} == osx ]] && echo "prepend-path DYLD_LIBRARY_PATH \$::env(PYTHIA6_ROOT)/lib")
```

Comment:
Ok, apparently I explained it badly. AGILe uses an environment variable to know where to look for generators (in addition to standard locations), this variable is AGILE_GEN_PATH. The path comes from the generator, and AGILe can work with different versions of the generator. So, every (relevant) generator should add itself to AGILE_GEN_PATH such that it can be picked up by AGILe. It's a bit like every package that wants to be executable adds itself to PATH.

If we added this to AGILe, and made it depend it on Pythia, we would have fixed it to one specific version to Pythia, or the other way around we would need another AGILe version for every generator.

Does this make sense now?